lua/plenary/log.lua

```diff
@@ -132,8 +132,20 @@ log.new = function(config, standalone)
     local info = debug.getinfo(config.info_level or 2, "Sl")
     local src_path = info.source:sub(2)
     local src_line = info.currentline
+    -- Notify
+    if config.use_console == "notify" then
+      local level = vim.log.levels[level_config.name:upper()]
+      if level == nil then
+        level = vim.log.levels.ERROR
+      end
```

Comment:
Why ERROR by default?  When can `level` be nil?

Reply:
plenary's log config defines levels like:
```lua
  modes = {
    { name = "trace", hl = "Comment" },
    { name = "debug", hl = "Comment" },
    { name = "info", hl = "None" },
    { name = "warn", hl = "WarningMsg" },
    { name = "error", hl = "ErrorMsg" },
    { name = "fatal", hl = "ErrorMsg" },
  },
```

while `vim.log.levels` is defined as:
```lua
    vim.log.levels.DEBUG
    vim.log.levels.ERROR
    vim.log.levels.INFO
    vim.log.levels.TRACE
    vim.log.levels.WARN
    vim.log.levels.OFF
```

These _almost_ match, but plenary defines a level `"fatal"` that `vim.log.levels` does not. I figured `vim.log.levels.ERROR` was the closest analogue to it, but maybe it's not sensible to just assume any unknown level should be treated as such. Thoughts?

Reply:
I wasn't aware of "fatal" in plenary. Makes sense, thanks for the explanation!